### DISCUSSION TITLE
fullySpecified: false does not work to resolve imports

### DISCUSSION THREAD

**Discussion by TheDirigible** [webpack/webpack#15027] [Q&A]
# Bug report

This error, first of all has awful grammar:

Module not found: Error: Can't resolve 'xxx' in 'yyy'
Did you mean 'xxx.js'?
BREAKING CHANGE: The request '...' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

Is the point of webpack to make development easier?  The developer does not want to specify the extension, and Webpack can DO THIS FOR HIM.  In fact, it used to.  This line "Did you mean 'xxx.js'?" shows that you do know exactly what he means.

There is documentation that fullySpecified:false makes this error go away, but it does not.

**What is the current behavior?**

failure to resolve

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

Imports should resolve.

**Other relevant information:**
webpack version:  5.65.0
Node.js version: 14.17.1
Operating System: Ubuntu 20
Additional tools:

**Comment by alexander-akait** [comment]
You can use only URLs in import, so you should add extensions, it is in the specification

**Reply by TheDirigible** [reply]
I understand the spec, however we are under no obligation to copy the spec.  Webpack is an enhancement over the basic spec.
Thousands of people have been using no extensions for 10+ years, now we want to force them all to convert?

Besides that, the fullySpecified flag, which should be false by default, was created to allow this.  It seems to be broken.

**Reply by alexander-akait** [reply]
Sorry, it is breaking change and yes you need to be align with spec and use `extension` on the end of `import`

**Reply by TheDirigible** [reply]
Why don’t you let some other people weigh in on this. Your responses are ignoring my points which means your mind is closed.

**Reply by alexander-akait** [reply]
I don't have problems with my mind, it is breaking change, we want to be align with ECMA spec

**Reply by alexander-akait** [reply]
Also we provide options to change behavior, if you need old behavior please use it

**Comment by TheDirigible** [comment]
If you read what I wrote, the option to restore correct previous behavior
is broken.

**Reply by alexander-akait** [reply]
You don't provide example of the problem, so I can't say why it is not working in your case, sorry I am not mage
